fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/QueryKeyExpression.java

```diff
@@ -326,24 +332,86 @@ public int planHash(@Nonnull final PlanHashMode mode) {
         public int queryHash(@Nonnull final QueryHashKind hashKind) {
             return HashUtils.queryHash(hashKind, SIMPLE_COMPARISON_BASE_HASH, super.queryHash(hashKind), getKeyExpression());
         }
+
+        @Nonnull
+        @Override
+        public Comparisons.Comparison withType(@Nonnull final Comparisons.Type newType) {
```

Comment:
This comment applies to the planHash here which seems to use `SIMPLE_COMPARISON_BASE_HASH` but should define its own I think.

Reply:
It's `QueryKeyExpressions. SIMPLE_COMPARISON_BASE_HASH`, not `SimpleComparisonBase. BASE_HASH`, so it already does define its own. They had to be in the parent class because these classes didn't used to be `static` and Java limitations. But now that they are `static`, they can be (renamed and) moved down.